integration/oasis/osrmconnector/connector.go

```diff
@@ -0,0 +1,45 @@
+package osrmconnector
+
+import (
+	"github.com/Telenav/osrm-backend/integration/pkg/api/osrm/route"
+)
+
+// RouteResponse contains osrm Route response and error
+type RouteResponse struct {
+	Resp *route.Response
+	Err  error
+}
+
+// OsrmConnector wraps the communication with OSRM server
+type OsrmConnector struct {
+	osrmclient *osrmHTTPClient
+}
+
+// NewOsrmConnector create OsrmConnector object
+func NewOsrmConnector(osrmEndpoint string) *OsrmConnector {
+	osrm := &OsrmConnector{
+		osrmclient: newOsrmHTTPClient(osrmEndpoint),
+	}
+	go osrm.osrmclient.start()
+	return osrm
+}
+
+// Request4Route provide async api to request for route
+func (oc *OsrmConnector) Request4Route(r *route.Request) <-chan RouteResponse {
```

Comment:
Why need these async api? How will you exepct to use them?

Reply:
When user calling `Request4Route`, he will get a channel of `RouteResponse` immediately. 
-  User could continue with other logic.  He will be blocked if he tries to fetch response from channel and the response is not ready yet.
- Inside osrmconnector, will communicate with OSRM server via http and assemble the response, when everything is finished then put result into response channel.
- Response contains `error` and actual response.  

I will update the upper information into doc.go.   The place where will use it would be:
- During logic, user might send several request for different range of osrm-table request, user will follow its own's logic to digest response.
- Search request is similar, user send request to sever side but could continue his own logic 
- I think design `promise-future` like api for `IO` related request is needed.  May be I should remove the word of `async`.